compiler/src/iree/compiler/DispatchCreation/FusionUtils.cpp

```diff
@@ -97,4 +100,44 @@ bool areFusableAsElementwiseOps(MLIRContext *context, OpOperand *fusedOperand,
   return true;
 }
 
+LogicalResult moveOperandDefs(RewriterBase &rewriter,
+                              ArrayRef<Operation *> operations,
+                              Operation *insertionPoint,
+                              DominanceInfo &dominanceInfo,
+                              ArrayRef<Operation *> ignoreOperations) {
+  BackwardSliceOptions options;
+  llvm::DenseSet<Operation *> ignoreOperationsSet;
+  ignoreOperationsSet.insert(ignoreOperations.begin(), ignoreOperations.end());
+  options.filter = [&](Operation *op) {
+    return !dominanceInfo.properlyDominates(op, insertionPoint) &&
+           !ignoreOperationsSet.contains(op);
+  };
+  // Set inclusive to true cause the slice is computed from the operand, and
+  // we want to include the defining op (which is the point here)
+  options.omitUsesFromAbove = false;
+  options.inclusive = true;
+
+  llvm::SetVector<Operation *> slice;
+  for (auto op : operations) {
+    for (auto operand : op->getOperands()) {
+      getBackwardSlice(operand, &slice, options);
+    }
+    auto regions = op->getRegions();
+    if (regions.empty()) {
+      continue;
+    }
+    llvm::SetVector<Value> capturedVals;
+    mlir::getUsedValuesDefinedAbove(regions, capturedVals);
+    for (auto value : capturedVals) {
+      getBackwardSlice(value, &slice, options);
+    }
+  }
```

Comment:
I think the following should be equivalent and shorter
 
```suggestion
  options.omitUsesFromAbove = false;
  options.inclusive = false;

  llvm::SetVector<Operation *> slice;
  for (auto op : operations) {
    getBackwardSlice(op, &slice, options);
  }
```

Reply:
Actually that is what I started with, but it didnt work. So I had to do this for it to work. I suspect that is not enough either.